xonsh/environ.py

```diff
@@ -1325,17 +1328,22 @@ def windows_foreign_env_fixes(ctx):
     ctx['PWD'] = _get_cwd()
 
 
+def foreign_env_fixes(ctx):
+    """Environment fixes for all operating systems"""
+    if 'PROMPT' in ctx:
+        del ctx['PROMPT']
+
+
 def default_env(env=None, config=None, login=True):
     """Constructs a default xonsh environment."""
     # in order of increasing precedence
     ctx = dict(BASE_ENV)
     ctx.update(os.environ)
-    if ON_WINDOWS:
-        # Windows style PROMPT definitions don't work in XONSH:
-        try:
-            del ctx['PROMPT']
-        except KeyError:
-            pass
+    # other shells' PROMPT definitions generally don't work in XONSH:
```

Comment:
I think this is worth a mention in the PROMPT envvar docs because this will also prevent inheriting the prompt variable even if it was perfectly valid because we launched xonsh from within xonsh.  I definitely think the robustness is worth the trade off, but we should probably let people know.